lib/internal/vm.js

```diff
@@ -151,6 +152,7 @@ function internalCompileFunction(
   }
 
   registerImportModuleDynamically(result.function, importModuleDynamically);
+  maybeCacheSourceMap(filename, code, result.function, false, result.sourceURL, result.sourceMapURL);
```

Comment:
What’s the difference between a “source URL” and a “source map URL”? My naïve guess is that a source URL is a `file:` URL to the module itself, and a source map URL is either a `file:` URL to a `.map` file or a `data:` URL with the source map. I feel like these guesses are probably wrong, though, so these variables should at least get comments or perhaps get better names (since you reuse these names across many files).

Reply:
Your guess is correct. `SourceURL` is used to specify the URL of the eval-ed source, as described in https://tc39.es/source-map/#linking-evald-code-to-named-generated-code. However, it can be present in sources loaded by CJS/ESM loaders from filesystem as well. V8 will take this magic comment and report errors with it as the script resource name instead: https://github.com/nodejs/node/blob/f1949ac1aebc652bf17f9ba8b3ab075c6dd1e3f6/src/node_errors.cc#L62-L65. So it has to be used as cache key along with its filename.

The name, `source url`, is taken from the magic comments as:

```
//# sourceMappingURL=<url>
//# sourceURL=foo.js
```